tests/templates/spark-linear-regression-history-server-job.yaml

```diff
@@ -44,6 +38,23 @@ spec:
     labels:
       version: {{ .SparkVersion }}
     serviceAccount: {{ .ServiceAccount }}
+    env:
+      - name: AWS_ACCESS_KEY_ID
+        valueFrom:
+          secretKeyRef:
+            name: {{ .Params.AwsSecretName }}
+            key: AWS_ACCESS_KEY_ID
+      - name: AWS_SECRET_ACCESS_KEY
+        valueFrom:
+          secretKeyRef:
+            name: {{ .Params.AwsSecretName }}
+            key: AWS_SECRET_ACCESS_KEY
+      - name: AWS_SESSION_TOKEN
+        valueFrom:
+          secretKeyRef:
+            name: {{ .Params.AwsSecretName }}
```

Comment:
I think `.Params.AwsSecretName` contradicts the approach described in https://github.com/kudobuilder/operators/pull/214, doesn't it?

Reply:
No, it is actually two different approaches: the one described here https://github.com/kudobuilder/operators/pull/214 is based on mounting a `spark-defaults.conf`, which contains `spark.hadoop.fs.s3a.*` properties, that will be picked up by SHS.
This one reflects the way we currently inject secrets into `SparkApplication` because of [this issue](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/issues/216) - as for now,  it is possible to provide Spark properties only via `spec.sparkConf`, but it will cause exposure of sensitive data in the application spec, so we are using a separate secret to provide keys and passwords to Spark app, e.g:
```yaml
apiVersion: v1
data:
  AWS_ACCESS_KEY_ID: <base64-encoded>
  AWS_SECRET_ACCESS_KEY: <base64-encoded>
kind: Secret
metadata:
  name: spark-secrets
type: Opaque
```

Reply:
Thanks for the clarification. My point here was more about why do we need an Operator parameter if we have to specify it on per-job basis? As this parameter is not applied automatically for all submitted jobs - there doesn't seem to be much sense of including it to into Operator configuration.

Reply:
They are two different parameters: `historyServerSparkConfSecret ` used in SHS only, and this `.Params.AwsSecretName ` is name of a secret used for Spark job in the test. Ideally, we should have I single Secret with spark properties, which would be shared between all the components (SHS, SparkApp), but as I said, currently there is no way to pass spark configuration to `SparkApplication` via conf file, so that’s why we create a separate secret with `AWS_*` credentials and use in Spark jobs

Reply:
Got it, thanks! So it is a parameter for tests only which makes sense.